src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp

```diff
@@ -278,6 +279,8 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
         // Ignore nodes that are not related to FullyConnected and allow ConstantFolding to be applied to them
         pass_config->set_callback<ov::pass::MarkDequantizationSubgraph>(is_non_decompression_multiply);
 
+        manager.register_pass<ov::intel_gpu::MoveConvertAfterGather>();
```

Comment:
Shouldn't this transformation be applied after `CommonOptimizations` ? I've tried this PR for qwen and it was not applied as pattern before folding looks different as dictionary is int8 compressed

Reply:
This case will be handled by this PR (https://github.com/openvinotoolkit/openvino/pull/21711).